catalog/cataloger_diff.go

```diff
@@ -414,48 +420,35 @@ func createDiffResultsTable(ctx context.Context, executor sq.Execer) (string, er
 	return diffResultsTableName, nil
 }
 
-func evaluateFromChildElementDiffType(effectiveCommits *diffEffectiveCommits, parentBranchID int64, childEnt *DBScannerEntry, parentEnt *DBScannerEntry) DifferenceType {
-	var matchedParent *DBScannerEntry
-	if parentEnt != nil && parentEnt.Path == childEnt.Path {
-		matchedParent = parentEnt
+func evaluateFromChildElementDiffType(effectiveCommits *diffEffectiveCommits, childEnt *DBScannerEntry, matchedParent *DBScannerEntry) DifferenceType {
+	// both deleted - none
+	if childEnt.IsDeleted() && (matchedParent == nil || matchedParent.IsDeleted()) {
+		return DifferenceTypeNone
 	}
-	// when the entry was deleted
-	if childEnt.IsDeleted() {
-		if matchedParent == nil || matchedParent.IsDeleted() {
-			return DifferenceTypeNone
-		}
-		if matchedParent.BranchID == parentBranchID {
-			// check if parent did any change to the entity
-			if matchedParent.MinCommit > effectiveCommits.ParentEffectiveCommit {
-				return DifferenceTypeConflict
-			}
-			return DifferenceTypeRemoved
-		}
-		// check if the parent saw this entry at the time
-		if matchedParent.MinCommit >= effectiveCommits.ParentEffectiveCommitByBranchID(matchedParent.BranchID) {
-			return DifferenceTypeRemoved
-		}
+
+	// same entry - none
+	if matchedParent != nil && childEnt.IsDeleted() == matchedParent.IsDeleted() && childEnt.Checksum == matchedParent.Checksum {
 		return DifferenceTypeNone
 	}
-	// when the entry was not deleted
+
+	// at this point both are not the same and not both are not deleted
```

Comment:
hard to understand comment - to many not

Reply:
removed one not